src/core/pageContent/meta/PageMeta.tsx

```diff
@@ -63,8 +62,10 @@ export function PageMeta({ page, headComponent: Head }: PageMetaProps) {
         hrefLang={page?.language?.locale}
         href={page?.seo?.canonicalUrl}
       />
-      <link rel="canonical" href={canonicalUrl} />
-      <link rel="canonical" href={canonicalUrl} />
+      <link
+        rel="canonical"
+        href={`${origin}${currentLanguageCode.toLowerCase()}/${page.slug}`}
```

Comment:
Shouldn't we use page uri instead? That would already contain the locale and other parent paths.